Glimmer/Assets/Scripts/DetectionAura.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class DetectionAura : MonoBehaviour
+{
+    private Light light;
+    private SphereCollider playerCollider;
+    private PlayerMovementController movementController;
+
+    void Start()
+    {
+        this.light = gameObject.GetComponent<Light>();
+        this.playerCollider = gameObject.GetComponent<SphereCollider>();
+        this.movementController = gameObject.GetComponentInParent<PlayerMovementController>();
+    }
+
+    private void Update()
+    {
+        if (this.movementController != null)
+        {
+            Vector3 movementVector = this.movementController.GetMovementVector();
+
+            if (Input.GetKey(KeyCode.LeftShift))
+            {
+                this.CrouchAura(movementVector.magnitude);
```

Comment:
If the magnitude of the vector is used in every case, then make a reference to the magnitude, instaead of a reference to the Vector3